deps/rabbit/src/rabbit_stream_coordinator.erl

```diff
@@ -253,6 +271,23 @@ apply(_Meta, {start_replica_failed, StreamId, Node, Retries, Reply},
                 ?RESTART_TIMEOUT * Retries}],
               State#?MODULE{streams = Streams})
     end;
+apply(_Meta, {update_retention_failed, StreamId, Retention, Retries, Reply},
+      #?MODULE{streams = Streams0} = State) ->
+    rabbit_log:debug("rabbit_stream_coordinator: ~p update retention failed", [StreamId]),
+    case maps:get(StreamId, Streams0, undefined) of
+        undefined ->
+            {State, {error, not_found}, []};
+        #{reply_to := From} = SState  ->
+            Streams = Streams0#{StreamId => clear_stream_state(SState)},
+            reply_and_run_pending(
+              From, StreamId, ok, Reply,
+              [{timer, {pipeline,
```

Comment:
I'm not sure why it is wrapped in `{pipeline, []}`? what does this do?

Reply:
It adds the command to the pipeline command, but delayed instead of immediately. I guess we don't want to retry immediately?